src/Compilers/Core/Portable/ConstantValue.cs

```diff
@@ -439,6 +439,7 @@ public object Value
                 {
                     case ConstantValueTypeDiscriminator.Bad: return null;
                     case ConstantValueTypeDiscriminator.Null: return null;
+                    case ConstantValueTypeDiscriminator.DefaultLiteral: return null;
```

Comment:
>case ConstantValueTypeDiscriminator.DefaultLiteral: return null; [](start = 20, length = 64)

It feels unexpected that we ended up with a special discriminator for this feature. This pretty much the same as Nothing literal in VB and we managed to support it without special discriminator.